### DISCUSSION TITLE
Allow setting to use & respect user `.npmrc` config to fix language server installation issues

### DISCUSSION THREAD

**Discussion by jamischarles** [zed-industries/zed#20359] [Feature Requests]
## Problem Statement
_npm-based_ language servers fail to install with non-standard npm setups. ie: custom registry, self-signed certs etc. This affects eslint, prettier, etc.

## Solution Proposal
Provide a setting to specify location of `.npmrc` to use with `--userconfig` flag

**Pros**
* Allows users to manually override zed to respect their global `.npmrc` settings which help set CA files, custom registry details etc
* Use that instead of the blank npmrc [that's being force overridden in the node runtime](https://github.com/zed-industries/zed/blob/main/crates/node_runtime/src/node_runtime.rs#L328)

**Cons**
* users may expect global .npmrc to already be respected. 

In this case IMO the pro far outweighs the con because there's no easy override (as far as I can tell)

## Open Questions
I spent a few hours last night building Zed on macOS and was finally able to manually override this in the code.

**Big questions I have remaining**:
1. I [can see that npmrc is completely being ignored right now](https://github.com/zed-industries/zed/blob/main/crates/node_runtime/src/node_runtime.rs#L328). The PR that added this doesn't provide much context on the reason. Am I reading that code right? That it's intentionally overriding `--userconfig` with a blank file?

## Related Issues
https://github.com/zed-industries/zed/issues/19806
https://github.com/zed-industries/zed/issues/9422
https://github.com/zed-industries/zed/issues/8650

**Comment by jamischarles** [comment]
Or would it be better to just allow people to opt-out of the [ignore user npmrc](https://github.com/zed-industries/zed/blob/main/crates/node_runtime/src/node_runtime.rs#L328) behavior that's currently the default?

**Comment by maximvl** [comment]
With a number of security issues and recent attacks through `npm` in corporate world it becomes increasingly popular to switch to alternatives like `pnpm` to a point that `npm` commands just won't work. I hope Zed can manage to support that

**Comment by yeskunall** [comment]
This should be supported now that https://github.com/zed-industries/zed/pull/26209 is merged. Feel free to open an issue if it doesn’t work as expected.
